decidim-core/app/controllers/decidim/data_portability_controller.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require "zip"
+
+module Decidim
+  # The controller to handle the user's download_my_data page.
+  class DataPortabilityController < Decidim::ApplicationController
+    include Decidim::UserProfile
+
+    def show
+      enforce_permission_to :show, :user, current_user: current_user
+
+      @account = form(AccountForm).from_model(current_user)
+    end
+
+    def export
+      enforce_permission_to :export, :user, current_user: current_user
+
+      DataPortabilityExportJob.perform_later(current_user, export_format)
+
+      flash[:notice] = t("decidim.account.data_portability_export.notice")
+
+      redirect_back(fallback_location: data_portability_path)
+    end
+
+    def download_file
+      enforce_permission_to :download, :user, current_user: current_user
+
+      if params[:token].present?
+        file_reader = Decidim::DataPortabilityFileReader.new(current_user, params[:token])
+        if file_reader.valid_token?
+          file = file_reader.file_path
+          if File.exist?(file)
```

Comment:
You'll find that data locality issue I pointed out before - the file might not exist here if the initial request was served on a process that lives potentially in another server. We should find a different way to do this - carrierwave looks like the way to go.